jdbc/src/main/java/io/kestra/jdbc/JdbcWorkerJobQueueService.java

```diff
@@ -0,0 +1,109 @@
+package io.kestra.jdbc;
+
+import io.kestra.core.exceptions.DeserializationException;
+import io.kestra.core.queues.QueueFactoryInterface;
+import io.kestra.core.queues.QueueInterface;
+import io.kestra.core.repositories.WorkerInstanceRepositoryInterface;
+import io.kestra.core.runners.*;
+import io.kestra.core.utils.Either;
+import io.kestra.jdbc.repository.AbstractJdbcWorkerJobRunningRepository;
+import io.kestra.jdbc.runner.JdbcHeartbeat;
+import io.kestra.jdbc.runner.JdbcQueue;
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.inject.qualifiers.Qualifiers;
+import jakarta.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.function.Consumer;
+
+@Singleton
+@Slf4j
+public class JdbcWorkerJobQueueService {
+    private final JdbcQueue<WorkerJob> workerTaskQueue;
+    private final JdbcHeartbeat jdbcHeartbeat;
+    private final AbstractJdbcWorkerJobRunningRepository jdbcWorkerJobRunningRepository;
+    private final WorkerInstanceRepositoryInterface workerInstanceRepository;
+    private Runnable queueStop;
+    private WorkerInstance workerInstance;
+
+    @SuppressWarnings("unchecked")
+    public JdbcWorkerJobQueueService(ApplicationContext applicationContext) {
+        this.workerTaskQueue = (JdbcQueue<WorkerJob>) applicationContext.getBean(
+            QueueInterface.class,
+            Qualifiers.byName(QueueFactoryInterface.WORKERJOB_NAMED)
+        );
+        this.jdbcHeartbeat = applicationContext.getBean(JdbcHeartbeat.class);
+        this.jdbcWorkerJobRunningRepository = applicationContext.getBean(AbstractJdbcWorkerJobRunningRepository.class);
+        this.workerInstanceRepository = applicationContext.getBean(WorkerInstanceRepositoryInterface.class);
+    }
+
+    public Runnable receive(String consumerGroup, Class<?> queueType, Consumer<Either<WorkerJob, DeserializationException>> consumer) {
+        workerInstance = jdbcHeartbeat.get();
+
+        this.queueStop = workerTaskQueue.receiveTransaction(consumerGroup, queueType, (dslContext, eithers) -> {
+            workerInstance = jdbcHeartbeat.get();
```

Comment:
You fetched twice the worker instance which is not needed, only fetch it when needed! The first `get` can be removed I think.

Reply:
It was need to be able to delete the instance when no job is receive.